### PR TITLE
Make case sensitivity optional

### PR DESCRIPTION
**Pull request by caesarxuchao**
Fix https://github.com/kubernetes/kubernetes/issues/64612

### CODE REVIEW

**Comment by caesarxuchao** [comment]
I tried unit tests on the master branch and many tests were broken. I suspect it's caused by dependency issues? I used "dep ensure" to download dependencies. It's much appreciated If you have some pointers on how to solve the local unit test failures. 

go test ./... output: https://raw.githubusercontent.com/caesarxuchao/go/test/testlog

**Comment by sai-g** [comment]
@caesarxuchao I faced the similar issue in running tests. I used the following command to fetch test dependencies, and it worked `go get -t ./...`

**Comment by caesarxuchao** [comment]
Thanks @sai-g. I also installed the json-iter/go at the wrong path. After moving it to $GOPATH/github.com/json-iterator/go, and run `go get -t ./...`, I'm able to pass the test locally.

File: reflect_struct_decoder.go [hunk 1]
```diff
@@ -51,7 +52,7 @@ func createStructDecoder(ctx *ctx, typ reflect2.Type, fields map[string]*structF
 		return &skipObjectDecoder{typ}
 	case 1:
 		for fieldName, fieldDecoder := range fields {
-			fieldHash := calcHash(fieldName)
+			fieldHash := calcHash(fieldName, ctx.caseSensitive())
```
Comment (liggitt) [inline reflect_struct_decoder.go]: Does this code above also make decoding case-insensitive?
``` 
for k, binding := range bindings {
  if _, found := fields[strings.ToLower(k)]; !found {
    fields[strings.ToLower(k)] = binding.Decoder.(*structFieldDecoder)
  }
}
```

Comment (liggitt) [reply reflect_struct_decoder.go]: A test case that exercises positive and negative cases would be good

Comment (nikhita) [reply reflect_struct_decoder.go]: Opened https://github.com/json-iterator/go/pull/285

Comment (nikhita) [reply reflect_struct_decoder.go]: > Does this code above also make decoding case-insensitive?

Yes, this was the culprit.

**Pull request merged**